front/components/vaults/VaultLayout.tsx

```diff
@@ -47,9 +51,19 @@ export function VaultLayout({
     parentId,
   } = pageProps;
   const router = useRouter();
-  const isPrivateVaultsEnabled = owner.flags.includes(
-    "private_data_vaults_feature"
-  );
+
+  const { vaults, isVaultsLoading } = useVaultsAsAdmin({
+    workspaceId: owner.sId,
+    disabled: plan.limits.vaults.maxVaults === 0,
+  });
+
+  const isPrivateVaultsEnabled =
+    owner.flags.includes("private_data_vaults_feature") &&
+    plan.limits.vaults.maxVaults > 0;
+  const isLimitReached =
+    isVaultsLoading ||
```

Comment:
`isVaultLoading` ?

Reply:
"vaults" is empty while loading - I prefer to have the button appear once loaded and ok, instead of having it disappear when loaded and limit reached

Reply:
Fair. Then nitpicking here but could we remove `isVaultLoading` from this condition and have:
```
setShowVaultCreationModal={
    isLimitReached || isVaultLoading ? undefined : setShowVaultCreationModal
}
```

Reply:
Actually not needed anymore if we want to display the button in all cases but use a dialog instead https://github.com/dust-tt/dust/pull/7744#discussion_r1780542107